src/CounterInterceptor/ClientCounter.cs

```diff
@@ -0,0 +1,87 @@
+namespace TemporalioSamples.CounterInterceptor;
+public record ClientCounts
+{
+    public uint Executions { get; internal set; }
+
+    public uint Signals { get; internal set; }
+
+    public uint Queries { get; internal set; }
+
+    public override string ToString()
+    {
+        return
+                "\n\tTotal Number of Workflow Exec: " + Executions +
+                "\n\tTotal Number of Signals: " + Signals +
+                "\n\tTotal Number of Queries: " + Queries;
+    }
+}
+
+public class ClientCounter
```

Comment:
Why have this separate class? Why not just use `Dictionary<string, ClientCounts>` where it's needed? Do not use static state, store the dictionary on the interceptor and make it available to the creator of the interceptor.

Reply:
Refactored.